src/baal/utils/pytorch_lightning.py

```diff
@@ -1,33 +1,55 @@
 import sys
 import types
-from abc import ABC, abstractmethod
+from abc import ABC
 from collections.abc import Sequence
-from typing import Dict, Any
+from typing import Dict, Any, Optional
 
 import numpy as np
 import structlog
-from pytorch_lightning import Trainer, Callback
-from tqdm import tqdm
-
+import torch
 from baal.active import ActiveLearningDataset
 from baal.active.heuristics import heuristics
 from baal.modelwrapper import mc_inference
 from baal.utils.cuda_utils import to_cuda
 from baal.utils.iterutils import map_on_tensor
+from pytorch_lightning import Trainer, Callback, LightningDataModule
+from torch.utils.data import DataLoader
+from tqdm import tqdm
 
 log = structlog.get_logger('PL testing')
 
 
+class BaaLDataModule(LightningDataModule):
+    def __init__(self, active_dataset: ActiveLearningDataset, batch_size=1,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.active_dataset = active_dataset
+        self.batch_size = batch_size
+
+    def pool_dataloader(self):
+        """Create Dataloader for the pool of unlabelled examples."""
+        return DataLoader(self.active_dataset.pool,
+                          batch_size=self.batch_size, num_workers=4, shuffle=False)
+
+    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        if 'active_dataset' in checkpoint:
+            self.active_dataset.load_state_dict(checkpoint['active_dataset'])
+        else:
+            log.warning("'active_dataset' not in checkpoint!")
+
+    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict:
+        checkpoint['active_dataset'] = self.active_dataset.state_dict()
+        return checkpoint
+
+
```

Comment:
may be worth considering the new teardown hook for LightningDataModule?

Reply:
Can you expand on what would we add in this hook?

Reply:
Nevermind, I guess if there is no `setup` there is no need for a `teardown` ;)

Reply:
LGTM!